### ISSUE TITLE
Socket Not Writable Error

### ISSUE THREAD

**Issue by nickcurmi** [dustindclark/homebridge-pentair-intellicenter#7]
**Describe The Bug:**

Logs showing errors saying the socket is not writable among other things. Homebridge server tries to connect every 30 seconds and fails each time. Actions in both home app and homebridge UI/X appear to be selectable but do not perform any expected actions.
**To Reproduce:**

Happening consistently every 30 seconds. No known steps to reproduce, it just continually occurs each time it attempts to connect. the 5 min snippet of logs that I posted below. 
**Expected behavior:**

For actions to occur when feature tiles are selected.
**Logs:**

```
[08/03/2022, 13:20:00] [homebridge-pentair-intellicenter] IntelliCenter socket has been closed. Waiting 30 seconds and attempting to reconnect...
[08/03/2022, 13:20:00] [homebridge-pentair-intellicenter] Caught error in sendCommandNoWait for command {
  "command": "GetQuery",
  "queryName": "GetHardwareDefinition",
  "arguments": "CIRCUITS",
  "messageID": "8d715050-e7ed-429d-bf17-43b94a171e5e"
} Error: socket not writable
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:264:16
    at Promise._execute (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:103:10)
    at Telnet.send (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:216:12)
    at PentairPlatform.sendCommandNoWait (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:397:21)
    at PentairPlatform.discoverDeviceType (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:269:10)
    at PentairPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:258:10)
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:151:18
[08/03/2022, 13:20:30] [homebridge-pentair-intellicenter] Finished waiting. Attempting reconnect...
[08/03/2022, 13:20:31] [homebridge-pentair-intellicenter] IntelliCenter socket error has been detected. Socket will be closed. Cannot connect
[08/03/2022, 13:20:31] [homebridge-pentair-intellicenter] Connection to IntelliCenter failed. Check config: {
  "host": "192.168.1.45",
  "port": 6681,
  "negotiationMandatory": false,
  "timeout": 1500,
  "debug": true,
  "username": "c28nick@yahoo.com",
  "password": "N0v122016"
} Error: Cannot connect
    at Socket.<anonymous> (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:87:25)
    at Object.onceWrapper (node:events:639:28)
    at Socket.emit (node:events:520:28)
    at Socket._onTimeout (node:net:501:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
[08/03/2022, 13:20:31] [homebridge-pentair-intellicenter] IntelliCenter socket has been closed. Waiting 30 seconds and attempting to reconnect...
[08/03/2022, 13:20:31] [homebridge-pentair-intellicenter] Caught error in sendCommandNoWait for command {
  "command": "GetQuery",
  "queryName": "GetHardwareDefinition",
  "arguments": "CIRCUITS",
  "messageID": "2d22ec8b-a147-40c1-9fc2-a19cf1bab6d3"
} Error: socket not writable
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:264:16
    at Promise._execute (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:103:10)
    at Telnet.send (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:216:12)
    at PentairPlatform.sendCommandNoWait (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:397:21)
    at PentairPlatform.discoverDeviceType (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:269:10)
    at PentairPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:258:10)
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:151:18
[08/03/2022, 13:21:01] [homebridge-pentair-intellicenter] Finished waiting. Attempting reconnect...
[08/03/2022, 13:21:03] [homebridge-pentair-intellicenter] IntelliCenter socket error has been detected. Socket will be closed. Cannot connect
[08/03/2022, 13:21:03] [homebridge-pentair-intellicenter] Connection to IntelliCenter failed. Check config: {
  "host": "192.168.1.45",
  "port": 6681,
  "negotiationMandatory": false,
  "timeout": 1500,
  "debug": true,
  "username": "c28nick@yahoo.com",
  "password": "N0v122016"
} Error: Cannot connect
    at Socket.<anonymous> (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:87:25)
    at Object.onceWrapper (node:events:639:28)
    at Socket.emit (node:events:520:28)
    at Socket._onTimeout (node:net:501:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
[08/03/2022, 13:21:03] [homebridge-pentair-intellicenter] IntelliCenter socket has been closed. Waiting 30 seconds and attempting to reconnect...
[08/03/2022, 13:21:03] [homebridge-pentair-intellicenter] Caught error in sendCommandNoWait for command {
  "command": "GetQuery",
  "queryName": "GetHardwareDefinition",
  "arguments": "CIRCUITS",
  "messageID": "67766d9b-d30e-484e-84b5-635b3671b55c"
} Error: socket not writable
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:264:16
    at Promise._execute (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:103:10)
    at Telnet.send (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:216:12)
    at PentairPlatform.sendCommandNoWait (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:397:21)
    at PentairPlatform.discoverDeviceType (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:269:10)
    at PentairPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:258:10)
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:151:18
[08/03/2022, 13:21:33] [homebridge-pentair-intellicenter] Finished waiting. Attempting reconnect...
[08/03/2022, 13:21:34] [homebridge-pentair-intellicenter] IntelliCenter socket error has been detected. Socket will be closed. Cannot connect
[08/03/2022, 13:21:34] [homebridge-pentair-intellicenter] Connection to IntelliCenter failed. Check config: {
  "host": "192.168.1.45",
  "port": 6681,
  "negotiationMandatory": false,
  "timeout": 1500,
  "debug": true,
  "username": "c28nick@yahoo.com",
  "password": "N0v122016"
} Error: Cannot connect
    at Socket.<anonymous> (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:87:25)
    at Object.onceWrapper (node:events:639:28)
    at Socket.emit (node:events:520:28)
    at Socket._onTimeout (node:net:501:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
[08/03/2022, 13:21:34] [homebridge-pentair-intellicenter] IntelliCenter socket has been closed. Waiting 30 seconds and attempting to reconnect...
[08/03/2022, 13:21:34] [homebridge-pentair-intellicenter] Caught error in sendCommandNoWait for command {
  "command": "GetQuery",
  "queryName": "GetHardwareDefinition",
  "arguments": "CIRCUITS",
  "messageID": "06fe621b-e04f-46c7-8140-c635bedc85d5"
} Error: socket not writable
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:264:16
    at Promise._execute (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:103:10)
    at Telnet.send (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:216:12)
    at PentairPlatform.sendCommandNoWait (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:397:21)
    at PentairPlatform.discoverDeviceType (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:269:10)
    at PentairPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:258:10)
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:151:18
[08/03/2022, 13:22:04] [homebridge-pentair-intellicenter] Finished waiting. Attempting reconnect...
[08/03/2022, 13:22:06] [homebridge-pentair-intellicenter] IntelliCenter socket error has been detected. Socket will be closed. Cannot connect
[08/03/2022, 13:22:06] [homebridge-pentair-intellicenter] Connection to IntelliCenter failed. Check config: {
  "host": "192.168.1.45",
  "port": 6681,
  "negotiationMandatory": false,
  "timeout": 1500,
  "debug": true,
  "username": "c28nick@yahoo.com",
  "password": "N0v122016"
} Error: Cannot connect
    at Socket.<anonymous> (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:87:25)
    at Object.onceWrapper (node:events:639:28)
    at Socket.emit (node:events:520:28)
    at Socket._onTimeout (node:net:501:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
[08/03/2022, 13:22:06] [homebridge-pentair-intellicenter] IntelliCenter socket has been closed. Waiting 30 seconds and attempting to reconnect...
[08/03/2022, 13:22:06] [homebridge-pentair-intellicenter] Caught error in sendCommandNoWait for command {
  "command": "GetQuery",
  "queryName": "GetHardwareDefinition",
  "arguments": "CIRCUITS",
  "messageID": "5c79ca4c-f776-4167-887c-0db0e893cfde"
} Error: socket not writable
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:264:16
    at Promise._execute (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:103:10)
    at Telnet.send (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:216:12)
    at PentairPlatform.sendCommandNoWait (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:397:21)
    at PentairPlatform.discoverDeviceType (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:269:10)
    at PentairPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:258:10)
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:151:18
[08/03/2022, 13:22:36] [homebridge-pentair-intellicenter] Finished waiting. Attempting reconnect...
[08/03/2022, 13:22:37] [homebridge-pentair-intellicenter] IntelliCenter socket error has been detected. Socket will be closed. Cannot connect
[08/03/2022, 13:22:37] [homebridge-pentair-intellicenter] Connection to IntelliCenter failed. Check config: {
  "host": "192.168.1.45",
  "port": 6681,
  "negotiationMandatory": false,
  "timeout": 1500,
  "debug": true,
  "username": "c28nick@yahoo.com",
  "password": "N0v122016"
} Error: Cannot connect
    at Socket.<anonymous> (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:87:25)
    at Object.onceWrapper (node:events:639:28)
    at Socket.emit (node:events:520:28)
    at Socket._onTimeout (node:net:501:8)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
[08/03/2022, 13:22:37] [homebridge-pentair-intellicenter] IntelliCenter socket has been closed. Waiting 30 seconds and attempting to reconnect...
[08/03/2022, 13:22:37] [homebridge-pentair-intellicenter] Caught error in sendCommandNoWait for command {
  "command": "GetQuery",
  "queryName": "GetHardwareDefinition",
  "arguments": "CIRCUITS",
  "messageID": "91a28bdb-e02d-4b6e-80c4-8ca22aff56ab"
} Error: socket not writable
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:264:16
    at Promise._execute (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/debuggability.js:384:9)
    at Promise._resolveFromExecutor (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:518:18)
    at new Promise (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/bluebird/js/release/promise.js:103:10)
    at Telnet.send (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/node_modules/telnet-client/lib/index.js:216:12)
    at PentairPlatform.sendCommandNoWait (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:397:21)
    at PentairPlatform.discoverDeviceType (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:269:10)
    at PentairPlatform.discoverDevices (/usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:258:10)
    at /usr/local/lib/node_modules/homebridge-pentair-intellicenter/src/platform.ts:151:18
```

**Plugin Config:**

```json

```
            "ipAddress": "192.168.1.45",
            "username": "XXXXX@XXXXXX.com",
            "password": "XXXXX",
            "temperatureUnits": "F",
            "minimumTemperature": 35,
            "maximumTemperature": 105,
            "_bridge": {
                "username": "0E:3F:41:7C:13:87",
                "port": 32188
            },
            "platform": "PentairIntelliCenter"
**Screenshots:**

**Environment:**

* **Plugin Version**:  v2.0.2
* **Homebridge Version**:  v1.4.0
* **Node.js Version**:  v16.14.0
* **NPM Version**:  v8.3.1
* **Operating System**:  Raspbian GNU/Linux Buster
--

**Comment by nickcurmi** [comment]
@dustindclark I think I figured out the issue. I noticed when I was logging into my intellicenter app the "local" option wasn't showing up any more. So I went digging through advanced settings in the panel. I realized that the last two digits of the IP address had changed. Not sure how/why that happened. Haven't made any adjustments to my network or anything. So, I changed the IP address in the plugin settings, restarted Homebridge then restarted the Child Bridge. Came right back online and started working! Might be something to put in the notes in case this issue happens to others. I'll close this issue.
